### ISSUE TITLE
[TOFU Reproduction] Released Llama-3.2-1B-Instruct NPO forget10 checkpoint does not reproduce documented metrics

### ISSUE THREAD

**Issue by asad073-ui** [locuslab/open-unlearning#199]
### Information

- [x] The official example scripts
- [ ] My own modified scripts

### Tasks

- [x] An officially supported task
- [ ] My own task or dataset (give details below)

### Reproduction

<html>
<body>
<html><head></head><body><h2>Summary</h2><p>I evaluated the released TOFU NPO <code inline="">forget10</code> checkpoint for <code inline="">Llama-3.2-1B-Instruct</code> using the official Open-Unlearning evaluator.</p><p>This report concerns <strong>evaluation of the released checkpoint</strong>. I did not retrain the NPO model.</p><p>The released full and retain90 checkpoints reproduce the documented results closely under the same evaluator and environment. However, the released NPO <code inline="">forget10</code> checkpoint consistently does not reproduce the documented <code inline="">model_utility = 0.46</code> and <code inline="">forget_truth_ratio = 0.70</code> results.</p><h2>Published result</h2><p>The Open-Unlearning reproducibility documentation reports the following TOFU <code inline="">forget10</code> results for <code inline="">Llama-3.2-1B-Instruct</code>:</p>
Checkpoint/method | Forget quality | Model utility | Forget truth ratio
-- | -- | -- | --
Full/finetuned | 1.66e-21 | 0.60 | 0.48
Retain90 | 1.0 | 0.59 | 0.63
NPO forget10 | 0.02 | 0.46 | 0.70

<p>The complete NPO measurements were:</p><pre><code class="language-text">forget_quality:       0.00024128267657124152
model_utility:        0.4323739724618672
forget_truth_ratio:   0.6413989131591031
forget_Q_A_ROUGE:     0.18626730143086795
forget_Q_A_Prob:      0.2075160337827401
extraction_strength:  0.09633125320089464
privleak:            -52.367840286308365
</code></pre><p>The corresponding full checkpoint measurements were:</p><pre><code class="language-text">forget_quality:      3.9054713571083378e-22
model_utility:       0.6001046250207063
forget_truth_ratio:  0.4753644629178423
</code></pre><p>The corresponding retain90 measurements were:</p><pre><code class="language-text">forget_quality:      0.999999999999994
model_utility:       0.5901844512864824
forget_truth_ratio:  0.6273686349110743
</code></pre><p>Therefore, under the same evaluator and environment:</p><ul><li><p>The full checkpoint closely reproduces <code inline="">0.60 / 0.48</code>.</p></li><li><p>The retain90 checkpoint closely reproduces <code inline="">0.59 / 0.63</code>.</p></li><li><p>The NPO checkpoint produces approximately <code inline="">0.432 / 0.641</code>, rather than <code inline="">0.46 / 0.70</code>.</p></li></ul><h2>Additional confirmation</h2><p>I also evaluated the same immutable NPO checkpoint using a separately pinned evaluation setup. That run produced:</p><pre><code class="language-text">model_utility:       0.43169057524720517
forget_truth_ratio:  0.6413989131591031
forget_quality:      0.00024128267657124152
</code></pre><p>The NPO mismatch therefore persisted across two evaluation setups, while the full and retain90 controls reproduced closely.</p><h2>Checks already performed</h2><ul><li><p>Pinned every Hugging Face model to an immutable revision.</p></li><li><p>Used the documented batch size of 32.</p></li><li><p>Used seed 0.</p></li><li><p>Used <code inline="">bfloat16</code>.</p></li><li><p>Used FlashAttention 2.</p></li><li><p>Evaluated on an RTX 3090 with native BF16 support.</p></li><li><p>Used unique task names and output directories.</p></li><li><p>Enabled evaluation overwrite.</p></li><li><p>Generated and supplied the retain90 evaluation logs.</p></li><li><p>Repeated the full checkpoint evaluation.</p></li><li><p>Evaluated the NPO checkpoint under the historical dependency stack.</p></li><li><p>Invoked the historical upstream evaluator directly without a compatibility shim.</p></li><li><p>Confirmed that the full and retain90 control checkpoints reproduce closely.</p></li></ul><h2>Clarification requested</h2><p>Could you please confirm whether the released NPO checkpoint corresponds to the checkpoint used to produce the documented TOFU <code inline="">forget10</code> row?</p><p>In particular, could you provide:</p><ol><li><p>The exact Hugging Face artifact and immutable revision used for the documented result.</p></li><li><p>The exact Open-Unlearning commit used for its evaluation.</p></li><li><p>The complete evaluation command and Hydra overrides.</p></li><li><p>The checkpoint epoch or training step that was evaluated.</p></li><li><p>Whether the documented row was generated directly from the currently released checkpoint.</p></li><li><p>Any additional dataset revision, retain-log, batching, precision, generation, or metric settings required for reproduction.</p></li></ol><p>At present, I am treating this as an unresolved released-artifact or evaluation-configuration mismatch. I am not concluding that the documented result is incorrect.</p></body></html>
</body>
</html>

[full_agg_values.json](https://github.com/user-attachments/files/30849362/full_agg_values.json)
[full_TOFU_SUMMARY.json](https://github.com/user-attachments/files/30849364/full_TOFU_SUMMARY.json)
[npo_agg_values.json](https://github.com/user-attachments/files/30849366/npo_agg_values.json)
[npo_TOFU_SUMMARY.json](https://github.com/user-attachments/files/30849367/npo_TOFU_SUMMARY.json)
[README.md](https://github.com/user-attachments/files/30849363/README.md)
[retain90_agg_values.json](https://github.com/user-attachments/files/30849368/retain90_agg_values.json)
[retain90_TOFU_SUMMARY.json](https://github.com/user-attachments/files/30849365/retain90_TOFU_SUMMARY.json)

### Expected behavior

The released NPO `forget10` checkpoint for `Llama-3.2-1B-Instruct`, when evaluated using the documented Open-Unlearning configuration, should reproduce the documented TOFU metrics within reasonable numerical tolerance.

Alternatively, the repository should identify the exact checkpoint artifact, immutable revision, evaluation commit, and configuration used to generate the documented `model_utility = 0.46` and `forget_truth_ratio = 0.70` result.

Because the full and retain90 control checkpoints reproduce closely under the same historical evaluator and environment, I expected the released NPO checkpoint to reproduce its corresponding documented row as well.
